lib/ZnapZend/ZFS.pm

```diff
@@ -12,12 +12,12 @@ use Data::Dumper;
 ### For "usual" datasets (filesystem,volume) `zfs` returns the
 ### properties inherited from higher level datasets; but for
 ### snapshots it only returns the same - not from higher snaps.
-my %inheritLevels = {
+my %inheritLevels = (
     local_only => 0,
     local_zfsinherit => 1,
     local_recurseparent => 2,
     local_recurseparent_zfsinherit => 3,
-};
+);
```

Comment:
as you mentioned :) sort of semi satisfying this way ...

Reply:
This PR is stalling and growing too large to review well already :( Can we leave such refinements for later iterations? I believe after the "proper fix" functionality would be the same, maintainability (key words instead of magic numbers) too, so that classifies as a refactor and can be separated :)

Reply:
Sure if you would like to follow up with another one to fix this properly ... I am fine with merging this one now. 

The reason I am trying to catch these things is that they are non-ideomatic and tend to make my life harder when I have to figure out some odd behavior some time down the road. As I tend to be the one stuck with these projects in the long run.